In [4]:
import pandas as pd 

In [7]:
data_2022 = pd.read_csv('../Data/diem_thi_thpt_2022.csv')
data_2023 = pd.read_csv('../Data/diem_thi_thpt_2023.csv')

In [22]:
data_2022.head()

sbd  toan  ngu_van  ngoai_ngu  vat_li  hoa_hoc  sinh_hoc  lich_su  \
0  1000001   3.6     5.00        4.0     NaN      NaN       NaN     2.75   
1  1000002   8.4     6.75        7.6     NaN      NaN       NaN     8.50   
2  1000003   5.8     7.50        5.0     NaN      NaN       NaN     7.25   
3  1000004   7.4     7.50        8.6     NaN      NaN       NaN     7.50   
4  1000005   7.2     8.50        9.0     NaN      NaN       NaN     8.00   

   dia_li  gdcd  
0     6.0  8.75  
1     7.5  8.25  
2     5.5  8.75  
3     6.5  7.50  
4     8.5  8.25

In [23]:
data_2023.head()

sbd  toan  ngu_van  ngoai_ngu  vat_li  hoa_hoc  sinh_hoc  lich_su  \
0  1000001   8.4     8.50        9.2     NaN      NaN       NaN     6.75   
1  1000002   7.2     8.50        9.2     NaN      NaN       NaN     8.75   
2  1000003   NaN     6.50        NaN     NaN      NaN       NaN     9.25   
3  1000004   7.8     8.25        7.8     NaN      NaN       NaN     4.50   
4  1000005   7.2     8.00        7.8     NaN      NaN       NaN     4.75   

   dia_li  gdcd ma_ngoai_ngu  
0    6.00  9.00           N1  
1    6.50  8.50           N1  
2    7.50   NaN          NaN  
3    6.25  8.25           N1  
4    6.75  8.25           N1

In [25]:
data_2022.shape

(995441, 10)

In [26]:
data_2023.shape

(1022060, 11)

In [27]:
#Print list of columns of each DataFrame
data_2022.columns

Index(['sbd', 'toan', 'ngu_van', 'ngoai_ngu', 'vat_li', 'hoa_hoc', 'sinh_hoc',
       'lich_su', 'dia_li', 'gdcd'],
      dtype='object')

In [28]:
data_2023.columns

Index(['sbd', 'toan', 'ngu_van', 'ngoai_ngu', 'vat_li', 'hoa_hoc', 'sinh_hoc',
       'lich_su', 'dia_li', 'gdcd', 'ma_ngoai_ngu'],
      dtype='object')

Compare to 2022, in 2023 there are more participants: 1022060 and 995441 participants in 2022. 

It can be seen from the dataset that in 2023 there is one more column which represent for the **Language code**. So it can be understand that there is no change in the subject that this contest held.

Moreover, except the ID column, the remains got null-value by that we can understand that the participants don't have to do the test for all subject, they can choose some of these. 